test/e2e/argo-workflows/templates/cpu-stress.yaml

```diff
@@ -79,37 +79,29 @@ spec:
       inputs:
         parameters:
           - name: namespace
-      activeDeadlineSeconds: 200
+      activeDeadlineSeconds: 300
       script:
         image: mongo:4.4.1
         command: [mongo, "fake-datadog.{{inputs.parameters.namespace}}.svc.cluster.local/datadog"]
         source: |
           while (1) {
-            var point = db.series.find({
-              metric: "kubernetes.cpu.usage.total",
-              tags: {$all: ["kube_deployment:cpu-stress", "kube_container_name:cpu-stress"]}
-            }).limit(1).sort({$natural:-1})[0];
-
-            if (point) {
-              value = point.points[0][1] / 10e6
-              print("cpu value: " + value)
-              // Sampling is less accurate than cgroups, widening the acceptable range
-              if (value > 35 && value < 45) {
-                print("cpu value in target range")
-                break;
-              }
-            } else {
-                print("no kubernetes.cpu.usage.total metric reported")
+            var nb = db.series.find({
+            metric: "kubernetes.cpu.usage.total",
+            tags: { $all: ["kube_deployment:cpu-stress", "kube_container_name:cpu-stress"] },
+            "points.0.1": { $gt: 800000000, $lt: 1200000000 } }).count();
+            print("find: " + nb)
+            if (nb != 0) {
+              print("cpu value in target range")
+              break;
```

Comment:
This new algorithm has a slightly different behaviour than the previous one.
The previous version was checking that the **last** point of the time series was in the expected range whereas the new one is checking that **any** point of the time series is in the expected range.
As the CPU consumption isn’t supposed to vary over time, it shouldn’t make a big difference.

Reply:
As we retry regularly it should be fine indeed